pkg/util/hostname.go

```diff
@@ -101,17 +101,41 @@ func setHostnameProvider(name string) {
 	inventories.SetAgentMetadata("hostname_source", name)
 }
 
-// GetHostname retrieve the host name for the Agent, trying to query these
+// GetHostname retrieves the host name from GetHostnameData
+func GetHostname() (string, error) {
+	hostnameData, err := GetHostnameData()
+	return hostnameData.Hostname, err
+}
+
+// HostnameProviderConfiguration is the key for the hostname provider associated to datadog.yaml
+const HostnameProviderConfiguration = "configuration"
+
+// HostnameData contains hostname and the hostname provider
+type HostnameData struct {
+	Hostname string
+	Provider string
+}
+
+func setHostnameData(cacheHostnameKey string, hostname string, provider string) HostnameData {
```

Comment:
please add a short comment on this function to explain what it does

Reply:
nit: now that I re-read the code, `saveHostnameData` may be a more appropriate name. Just seems a bit weird that `GetHostnameData` has a lot of logic and then calls `setHostnameData` to cache the values it resolved

Reply:
Fixed.